### PR TITLE
[Projects] Raise error for workflow scheduling with non-remote project [1.2.x]

### PR DESCRIPTION
**Pull request by yonishelach**
[ML-2931](https://jira.iguazeng.com/browse/ML-2931)

### CODE REVIEW

**Comment by yonishelach** [comment]
The remote checking looks pretty tough to process, should I move it to `MlrunProject` class?

File: mlrun/projects/pipelines.py [hunk 1]
```diff
@@ -713,6 +713,24 @@ def run(
...
+            or any(project.spec.source.startswith(prefix) for prefix in [".", "/"])
+            or (
+                not project.spec.source.startswith("git://")
+                or not any(
+                    project.spec.source.endswith(suffix)
+                    for suffix in ["tar.gz", ".zip"]
+                )
+            )
+        ):
+            raise ValueError(
```
Comment (yonishelach) [inline mlrun/projects/pipelines.py:729]: not sure if the error message is clear enough and if it should be `ValueError`

**Review by AlonMaor14 — COMMENTED** [review]
Great addition! A few comments / questions.
Also, I would love an explanation for why is this not supported.

File: mlrun/projects/pipelines.py [hunk 2]
```diff
@@ -713,6 +713,24 @@ def run(
         runner_name = f"workflow-runner-{workflow_name}"
         run_id = None
 
+        if (
```
[the remark this answers is not in the document]
Comment (AlonMaor14) [reply mlrun/projects/pipelines.py:716]: Also, once you extract the method, please add a docstring of what counts as a remote project (git / archives)

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
This is missing tests, please add :)

**Review by Tankilevitch — APPROVED** [review]

**Pull request merged**